src/modules/75-cf/components/FlagArchiving/CannotArchiveWarning.tsx

```diff
@@ -6,50 +6,77 @@
+import css from './useArchiveFlagDialog.module.scss'
 
 export interface CannotArchiveWarningProps {
+  dependentFlagsResponse: Features
   flagIdentifier: string
-  prerequisites: Prerequisite[]
+  flagName: string
+  queryParams: GetDependentFeaturesQueryParams
+  pageNumber: number
+  setPageNumber: (pageNum: number) => void
 }
 
-const CannotArchiveWarning: FC<CannotArchiveWarningProps> = ({ flagIdentifier, prerequisites }) => {
+export const CannotArchiveWarning: FC<CannotArchiveWarningProps> = ({
+  dependentFlagsResponse,
+  flagName,
+  queryParams,
+  pageNumber,
+  setPageNumber
+}) => {
   const { getString } = useStrings()
   const { withActiveEnvironment } = useActiveEnvironment()
-  const { projectIdentifier, orgIdentifier, accountId } = useParams<Record<string, string>>()
 
   return (
     <Layout.Vertical spacing="small">
-      <Text style={{ wordBreak: 'break-word' }} font={{ variation: FontVariation.BODY2 }}>
-        {getString('cf.featureFlags.archiving.cannotArchive', { flagIdentifier })}:
-      </Text>
-      {prerequisites.map(flag => (
-        <ItemContainer data-testid="flag-prerequisite-row" key={flag.feature}>
+      <String
+        className={css.cannotArchiveMessage}
+        stringID="cf.featureFlags.archiving.cannotArchive"
+        useRichText
+        vars={{ flagName }}
+      />
+      {dependentFlagsResponse?.features?.map(flag => (
+        <ItemContainer data-testid="dependent-flag-row" key={flag.identifier}>
+          <Link
+            to={withActiveEnvironment(
+              routes.toCFFeatureFlagsDetail({
+                accountId: queryParams.accountIdentifier,
+                orgIdentifier: queryParams.orgIdentifier as string,
+                projectIdentifier: queryParams.projectIdentifier as string,
+                featureFlagIdentifier: flag.identifier
+              })
+            )}
+          >
+            {flag.name}
+          </Link>
           <Layout.Horizontal>
             <Text padding={{ right: 'xsmall' }}>{getString('common.ID')}:</Text>
-            <Link
-              to={withActiveEnvironment(
-                routes.toCFFeatureFlagsDetail({
-                  orgIdentifier: orgIdentifier as string,
-                  projectIdentifier: projectIdentifier as string,
-                  featureFlagIdentifier: flag.feature,
-                  accountId
-                })
-              )}
-            >
-              {flag.feature}
-            </Link>
+            <Text>{flag.identifier}</Text>
           </Layout.Horizontal>
         </ItemContainer>
       ))}
-      <Text font={{ variation: FontVariation.BODY2 }}>{getString('cf.featureFlags.archiving.removeFlag')}</Text>
+      <Text padding={{ top: 'medium' }} color={Color.GREY_500} font={{ variation: FontVariation.BODY2 }}>
+        {getString('cf.featureFlags.archiving.removeFlag')}
+      </Text>
+      <Pagination
+        gotoPage={pageNum => {
+          setPageNumber(pageNum)
+        }}
```

Comment:
Super-minor and deffo not a blocker, but you can simplify here
```suggestion
        gotoPage={setPageNumber}
```